storage/client_raft_test.go

```diff
@@ -382,7 +389,7 @@ func TestFailedReplicaChange(t *testing.T) {
 
 	// The pending config change flag was cleared, so a subsequent attempt
 	// can succeed.
-	storage.TestingCommandFilter = nil
+	runFilter.Store(false)
```

Comment:
Hmm, I've changed the test to use a `defer, but still got a race error:

https://circle-artifacts.com/gh/cockroachdb/cockroach/4137/artifacts/0/tmp/circle-artifacts.eLAGyKb/excerpt.txt

```
WARNING: DATA RACE
Read by goroutine 49:
  github.com/cockroachdb/cockroach/storage.(*Range).executeCmd()
      /go/src/github.com/cockroachdb/cockroach/storage/range_command.go:53 +0x4d9
  github.com/cockroachdb/cockroach/storage.(*Range).applyRaftCommand()
      /go/src/github.com/cockroachdb/cockroach/storage/range.go:888 +0xa0d
  github.com/cockroachdb/cockroach/storage.(*Range).processRaftCommand()
      /go/src/github.com/cockroachdb/cockroach/storage/range.go:776 +0x372
  github.com/cockroachdb/cockroach/storage.funcÂ·043()
      /go/src/github.com/cockroachdb/cockroach/storage/store.go:1492 +0xd5d
  github.com/cockroachdb/cockroach/util.funcÂ·001()
      /go/src/github.com/cockroachdb/cockroach/util/stopper.go:75 +0x7f

Previous write by goroutine 205:
  github.com/cockroachdb/cockroach/storage_test.funcÂ·032()
      /go/src/github.com/cockroachdb/cockroach/storage/client_range_test.go:301 +0x38
  github.com/cockroachdb/cockroach/storage_test.TestTxnPutOutOfOrder()
      /go/src/github.com/cockroachdb/cockroach/storage/client_range_test.go:311 +0x163e
  testing.tRunner()
      /usr/src/go/src/testing/testing.go:447 +0x133
```

Do I need to make the test wait all activities finish?